src/rules/non-interactive-tooltip-trigger.js

```diff
@@ -0,0 +1,126 @@
+const {isPrimerComponent} = require('../utils/is-primer-component')
+const {getJSXOpeningElementName} = require('../utils/get-jsx-opening-element-name')
+const {getJSXOpeningElementAttribute} = require('../utils/get-jsx-opening-element-attribute')
+
+isInteractive = child => {
+  const childName = getJSXOpeningElementName(child.openingElement)
+  return ['button', 'summary', 'select', 'textarea', 'a', 'input', 'iconbutton'].includes(childName.toLowerCase())
+}
+
+isInteractiveAnchor = child => {
+  const hasHref = getJSXOpeningElementAttribute(child.openingElement, 'href')
+  if (!hasHref) return false
+  const href = getJSXOpeningElementAttribute(child.openingElement, 'href').value.value
+  const isAnchorInteractive = typeof href === 'string' && href !== ''
+  return isAnchorInteractive
+}
+
+isInteractiveInput = child => {
+  const hasHiddenType =
+    getJSXOpeningElementAttribute(child.openingElement, 'type') &&
+    getJSXOpeningElementAttribute(child.openingElement, 'type').value.value === 'hidden'
+  return !hasHiddenType
+}
+
+const checkTriggerElement = jsxNode => {
```

Comment:
One idea for collecting errors and going through all children here could be a block like:

```suggestion
const checks = [
  {
    id: 'anchorTagWithoutHref',
    filter: (jsxElement) => {
      return jsxElement.openingElement.name === 'a';
    },
    check: isInteractiveAnchor,
  },
  {
    id: 'hiddenInput',
    filter: (jsxElement) => {
      return jsxElement.openingElement.name === 'input';
    },
    check: isInteractiveInput,
  },
];

const checkTriggerElement = jsxNode => {  
  const elements = [jsxNode, ...getAllChildren(jsxNode)];
  const hasInteractiveElement = elements.find(element => {
    return isInteractive(element);
  });
  
  if (hasInteractiveElement) {
    return;
  }
  
  const errors = new Set();
  
  for (const element of elements) {
    for (const check of checks) {
      if (!check.filter(element)) {
        continue;
      }
      
      if (!check.check(element)) {
        errors.add(check.id);
      }
    }
  }
};
```

Basically get all the elements in this part of the JSX tree, check to see if there is an interactive one, and if not try and find the relevant errors and report whatever the first one is.

Reply:
I like this 💖 It is cleaner. I had to play with it to cover all test cases but it helped a lot.